### ISSUE TITLE
[BUG] org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/30_discovery/Discovery stats}

### ISSUE THREAD

**Issue by shwetathareja** [opensearch-project/OpenSearch#10931]
**Describe the bug**
Discovery stats structure got updated in https://github.com/opensearch-project/OpenSearch/pull/10670
This is causing org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/30_discovery/Discovery stats} tests to fail

https://build.ci.opensearch.org/job/gradle-check/28987/

**Comment by shwetathareja** [comment]
@amkhar can you please take a look and update the test asap.

**Comment by shwetathareja** [comment]
Also check if other node stats test failures are related

[org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/20_response_filtering/Nodes Stats with response filtering}](https://build.ci.opensearch.org/job/gradle-check/28987/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_nodes_stats_20_response_filtering_Nodes_Stats_with_response_filtering_/)
[org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/20_response_filtering/Nodes Stats with response filtering}](https://build.ci.opensearch.org/job/gradle-check/28987/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_nodes_stats_20_response_filtering_Nodes_Stats_with_response_filtering__2/)
[org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/20_response_filtering/Nodes Stats with response filtering}](https://build.ci.opensearch.org/job/gradle-check/28987/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_nodes_stats_20_response_filtering_Nodes_Stats_with_response_filtering__3/)
[org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT.test {p0=nodes.stats/20_response_filtering/Nodes Stats with response filtering}](https://build.ci.opensearch.org/job/gradle-check/28987/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_nodes_stats_20_response_filtering_Nodes_Stats_with_response_filtering__4/)

**Comment by amkhar** [comment]
I'll take a look shortly. Thanks for reporting this.

**Comment by amkhar** [comment]
These mixed cluster test were failing even before [https://build.ci.opensearch.org/job/gradle-check/28795/testReport/junit/org.opensearch.b[…]iteIT/test__p0_nodes_stats_30_discovery_Discovery_stats_/](https://build.ci.opensearch.org/job/gradle-check/28795/testReport/junit/org.opensearch.backwards/MixedClusterClientYamlTestSuiteIT/test__p0_nodes_stats_30_discovery_Discovery_stats_/)
GC link : https://build.ci.opensearch.org/job/gradle-check/28795/
Also I checked the code specifically for discovery one

```  
- do:
      nodes.stats:
        filter_path: "nodes.*.discovery"

  - is_false: cluster_name
  - is_true:  nodes
  - is_false: nodes.$master.name
  - is_false: nodes.$master.jvm
  - is_true:  nodes.$master.discovery
```

I did this from code base.

```
curl localhost:9200/_nodes/stats?filter_path=nodes.*.discovery
{"nodes":{"Ubn8Zxo9R06G299on2XfCw":{"discovery":{"cluster_state_queue":{"total":0,"pending":0,"committed":0},"published_cluster_states":{"full_states":2,"incompatible_diffs":0,"compatible_diffs":0},"cluster_state_stats":{"overall":{"update_count":2,"total_time_in_millis":101,"failed_count":0}}}}}}
```

We’ll always get discovery section as nothing is changed in that.

After rebasing from main these failures will be gone as new version which is merged in main is 2.12.0
Please close this issue once you verify that other PRs are not facing this after merging from main.

**Comment by shwetathareja** [comment]
Closing this issue.
